components/tidb_query/src/codec/chunk/column.rs

```diff
@@ -289,6 +381,36 @@ impl Column {
         self.finished_append_var()
     }
 
+    /// Append a bytes in raw datum format to the column.
+    pub fn append_bytes_datum(&mut self, mut raw_datum: &[u8]) -> Result<()> {
+        if raw_datum.is_empty() {
+            return Err(Error::InvalidDataType(
+                "Failed to decode datum flag".to_owned(),
+            ));
+        }
+        let flag = raw_datum[0];
+        raw_datum = &raw_datum[1..];
+        match flag {
+            datum::NIL_FLAG => self.append_null(),
+            // In index, it's flag is `BYTES`. See TiDB's `encode()`.
+            datum::BYTES_FLAG => self.append_bytes(&raw_datum.read_datum_payload_bytes()?),
```

Comment:
please add a TODO for it so that we know that it's performance can be further improved